File: pkg/id.generator.ts
export function generateNumericId(length = 12) {
  let result: string = '';
  const chars = '0123456789';
  for (let i = 0; i < length; i++) {
    result += chars.charAt(Math.floor(Math.random() * chars.length));
  }
  return result;
}

File: src/main/java/com/matreshka/auth_service/internal/components/IdGenerator.java
package com.matreshka.auth_service.internal.components;

import org.springframework.stereotype.Component;

import java.util.Random;

@Component
public class IdGenerator {

    private static final String CHARS = "0123456789";
    private static final int LENGTH = 12;
    private final Random random = new Random();

    public String generateId() {
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < LENGTH; i++) {
            result.append(CHARS.charAt(random.nextInt(CHARS.length())));
        }
        return result.toString();
    }
}
